aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -803,6 +819,26 @@ Tensor& __ilshift__(Tensor& self, const Scalar& other) {
   return self;
 }
 
+TORCH_IMPL_FUNC(bitwise_left_shift_out) (const Tensor& self, const Tensor& other, const Tensor& result) {
```

Comment:
Let's talk about this behavior and the current shift functions.

PyTorch defines the following shift functions today: __lshift__, __ilshift__, __rshift__, __irshift__. And these and the current PR have the same type promotion bug, as shown by this sample program:

```
t = torch.tensor((1, 2, 3))
t << 3.2
: tensor([ 8, 16, 24])
```

The current shift functions and these new ones also have divergent dtype support, as shown here:

```
t = torch.tensor((1., 2, 3))
t << 3.2
: tensor([ 9.1896, 18.3792, 27.5688])
```

This creates a tricky situation. Here's what I'd like to propose we do:

- the new functions should be implemented as aliases of the existing shift functions plus the new reflective shifts (they should just call the existing functions)
- the incorrect type promotion behavior should be preserved, because changing it would be BC-breaking

This will preserve float support, which would also require a BC-breaking change, but it will ensure that PyTorch's shift functionality is consistent.

Reply:
@mruberry I agree to support floating-point type arguments to make the behavior consistent with existing `__lshift__`, and will fix the implementation!
However, `__lshift__` and `__ilshift__` don't have an interface for `bitwise_left_shift_out`, so the new functions cannot be implemented as aliases of existing functions. So I am planning to support `bitwise_left_shift_out` that calls `lshift_stub`. Is it OK?

Reply:
Absolutely, you're right. Sorry, I didn't mean to suggest that we not add any new code, just that we be sure to reuse the same behavior (and its natural extension)